src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -101,6 +101,10 @@ void BaseCouplingScheme::sendData(const m2n::PtrM2N &m2n, const DataMap &sendDat
   PRECICE_ASSERT(m2n->isConnected());
 
   for (const auto &data : sendData | boost::adaptors::map_values) {
+    const auto stamples = data->getStamples();
+    PRECICE_ASSERT(stamples.size() > 0);
+    data->sample() = stamples.back().sample;
```

Comment:
For me to understand better: Why don't we loop over all stamples here?

Reply:
This is something I would rather implement in a later pull request. Currently we only communicate the sample for `WINDOW_END`, because this is what we also did before. In the future we will serialize all stamples and send them to the other participant.